3_visualize/src/drought_learner_viz_components.R

```diff
@@ -0,0 +1,889 @@
+  
+  fill_drought <- bind_rows(fill_drought_bottom, fill_drought_top) %>%
+    mutate(id = 1)
+  
+  mask_poly <- data.frame(
+    id = rep(1, 8),
+    subid = rep(1:2,each = 4),
+    x = c(as.Date("01/05/1963",'%d/%m/%Y'), as.Date("01/05/1963",'%d/%m/%Y'), 
+          as.Date("15/10/1963",'%d/%m/%Y'), as.Date("15/10/1963",'%d/%m/%Y'),
+          as.Date("15/08/1963",'%d/%m/%Y'), as.Date("15/08/1963",'%d/%m/%Y'), 
+          as.Date("28/08/1963",'%d/%m/%Y'), as.Date("28/08/1963",'%d/%m/%Y')),
+    y = c(-Inf, Inf, Inf, -Inf,
+          0, 100, 100, 0)
+  )
+  
+  main <- blank_plot +
+    # This site's 1963 streamflow
+    geom_line(color = "#1e41b5", size = 0.3)+
+    annotate("text", label = "Daily\nstreamflow", 
+             x = as.Date("03/06/1963",'%d/%m/%Y'), hjust = 0,
+             y = 420, color = "#1e41b5", size = 2)+
+    # Average daily streamflow
+    geom_line(aes(y = mean_flow), color = "#c3e8fa")+
+    annotate("text", label = "Daily average\nstreamflow", 
+             x = as.Date("17/07/1963",'%d/%m/%Y'), hjust = 0,
+             y = 800, color = "#5691e2", size = 2)+
+    # Average daily streamflow across year
+    geom_line(aes(y = mean(mean_flow)), color = "#da7d81")+
+    # Severe drought threshold (10%)
+    geom_line(aes(y = thresh_10_site), color = "#c1272d")+
+    annotate("text", label = "10% of annual average streamflow", 
+             x = as.Date("17/05/1963",'%d/%m/%Y'), hjust = 0,
+             y = 15, color = "#c1272d", size = 2)+
+    # Transparent mask
+    geom_polygon(data = mask_poly, 
+                 aes(group = id, x = x, y = y, subgroup = subid), 
+                 fill = "white", alpha = 0.8)+
+    geom_polygon(data = mask_poly %>% filter(subid == 2), 
+                 aes(group = id, x = x, y = y), 
+                 color = "black", fill = "transparent", size = 0.1)+
+    annotate("segment", y = 100, yend = 400, 
+             x = as.Date("29/08/1963",'%d/%m/%Y'),
+             xend = as.Date("26/09/1963",'%d/%m/%Y'),
+             color = "black", size = 0.2)+
+    annotate("segment", y = 400, yend = 100, 
+             x = as.Date("16/07/1963",'%d/%m/%Y'),
+             xend = as.Date("14/08/1963",'%d/%m/%Y'),
+             color = "black", size = 0.2)
+  
+  # Zoomed in plot
+  zoom_plot <- ggplot(data = p2_streamflow_learner_viz_df, aes(y = value, x = dt))+
+    # This site's 1963 streamflow
+    geom_line(color = "#1e41b5", size = 0.3)+
+    annotate("text", label = "Daily\nstreamflow", 
+             x = as.Date("03/06/1963",'%d/%m/%Y'), hjust = 0,
+             y = 420, color = "#1e41b5", size = 2)+
+    # Severe drought threshold (10%)
+    geom_line(aes(y = thresh_10_site), color = "#c1272d")+
+    annotate("text", label = "10% of annual average streamflow", 
+             x = as.Date("17/05/1963",'%d/%m/%Y'), hjust = 0,
+             y = 15, color = "#c1272d", size = 2)+
+    # Severe drought fill
+    geom_polygon(data = fill_drought, aes(group = id, x = dt, y = value), 
```

Comment:
Note - this is _not_ a requested change, because the code works as-is and produces the desired chart for the meeting tomorrow, but I wonder if this fill effect could be achieved more simply using `geom_ribbon`? Might be something to explore when you revisit this code.

Reply:
Oooo, that might just work! and fix the minor issue of the corners of the drought fill not quite lining up.